goSrc/src/vmCaller/rpc/eth_service.go

```diff
@@ -0,0 +1,1034 @@
+		Transaction: getTransaction(head, hash, tx),
+	}, nil
+}
+
+// EthGetTransactionCount returns the number of transactions sent from an address
+func (srv *EthService) EthGetTransactionCount(req *web3.EthGetTransactionCountParams) (*web3.EthGetTransactionCountResult, error) {
+	addr, err := x.DecodeToAddress(req.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: get tx count at height
+	acc, err := srv.accounts.GetAccount(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: sequence may not always be accurate, is there a better way?
+	return &web3.EthGetTransactionCountResult{
+		NonceOrNull: x.EncodeNumber(acc.GetSequence()),
+	}, nil
+}
+
+func getHashAndCallTxFromEnvelope(env *txs.Envelope) ([]byte, *payload.CallTx, error) {
+	if env.Tx == nil {
+		return nil, nil, fmt.Errorf("tx not found for %s", env.String())
+	} else if tx, ok := env.Tx.Payload.(*payload.CallTx); ok {
+		return env.Tx.Hash().Bytes(), tx, nil
+	}
+	return nil, nil, fmt.Errorf("tx not valid")
+}
+
+func getHashAndCallTxFromExecution(txe *exec.TxExecution) ([]byte, *payload.CallTx, error) {
+	if txe.Envelope == nil {
+		return nil, nil, fmt.Errorf("envelope not found for %s", txe.GetTxHash().String())
+	}
+	return getHashAndCallTxFromEnvelope(txe.Envelope)
+}
+
+// EthGetTransactionReceipt returns the receipt of a previously committed tx
+func (srv *EthService) EthGetTransactionReceipt(req *web3.EthGetTransactionReceiptParams) (*web3.EthGetTransactionReceiptResult, error) {
+	data, err := x.DecodeToBytes(req.TransactionHash)
+	if err != nil {
+		return nil, err
+	}
+
+	txe, err := srv.events.TxByHash(data)
+	if err != nil {
+		return nil, err
+	} else if txe == nil {
+		return nil, fmt.Errorf("tx with hash %s does not exist", req.TransactionHash)
+	}
+
+	hash, tx, err := getHashAndCallTxFromExecution(txe)
+	if err != nil {
+		return nil, err
+	}
+
+	block, err := srv.blockchain.GetBlockHeader(txe.Height)
+	if err != nil {
+		return nil, err
+	}
+
+	status := x.EncodeNumber(1)
+	if err := txe.Exception.AsError(); err != nil {
+		status = x.EncodeNumber(0)
+	}
+
+	result := &web3.EthGetTransactionReceiptResult{
+		Receipt: web3.Receipt{
+			Status:            status,
+			TransactionIndex:  x.EncodeNumber(txe.GetIndex()),
+			BlockNumber:       x.EncodeNumber(uint64(block.Height)),
+			BlockHash:         x.EncodeBytes(block.Hash()),
+			From:              x.EncodeBytes(tx.GetInput().Address.Bytes()),
+			GasUsed:           x.EncodeNumber(txe.Result.GetGasUsed()),
+			TransactionHash:   x.EncodeBytes(hash),
+			CumulativeGasUsed: hexZero,
+			LogsBloom:         hexZero,
+			Logs:              []web3.Logs{},
+		},
+	}
+
+	if txe.Receipt != nil {
+		result.Receipt.ContractAddress = x.EncodeBytes(txe.Receipt.ContractAddress.Bytes())
+		result.Receipt.To = pending
+	} else if tx.Address != nil {
+		result.Receipt.To = x.EncodeBytes(tx.Address.Bytes())
+	}
+
+	return result, nil
+}
+
+// EthHashrate returns the configured tendermint commit timeout
```

Comment:
"tendermint" - the word does not exist, what do You mean?

Reply:
It's consensus algorithm used int burrow: https://tendermint.com/ check it out